internal/builders/bazel/rebuilder.sh

```diff
@@ -0,0 +1,490 @@
+    echo ""
+else
+    if [[ -n "$docker_image" ]]
+    then
+      # Warning message for the users if their artifact was not built with a Docker Image, but a Docker Image was provided at command.
+      printf "${RED}[Warning] ${LIGHT_RED}Docker Image, %s, provided, but artifact was not originally built on Docker Image${RESET}\n" "$docker_image"
+    else
+      echo "" # This is just for style.
+    fi
+
+    # Run the build script locally without a docker image.
+    printf "${CYAN}=============================================${RESET}\n"
+    type_writer "💻---> Rebuilding with local environment..."
+    printf "${CYAN}=============================================${RESET}\n"
+    echo ""
+
+    # shellcheck source=../build.sh
+    source ../build.sh
+    echo ""
+    printf "${CYAN}=============================================${RESET}\n"
+    printf "${CYAN}|${RESET}${YELLOW}${UNDERLINE}        ✅  Artifacts Rebuilt! ✅          ${RESET}${CYAN}|${RESET}\n"
+    printf "${CYAN}=============================================${RESET}\n"
+    echo ""
+fi
+
+# To avoid unbound variable after build script which sets -euo.
+set +u
+
+# If Docker Image was used to build on Github, we need to cd into repo
+# to access the binaries directory.
+if [[ -n $DOCKER_IMAGE ]]
+then
+  cd "$repo_name"
+fi
+
+################################################
+#                                              #
+#               Copy the Artifact              #
+#                                              #
+################################################
+
+# Obtain the name of the artifact
+if [[ $artifact_path == */* ]]
+then
+    artifact_name=$(basename "$artifact_path")
+else
+    artifact_name=$artifact_path
+fi
+
+rebuilt_checksum=""
+unset rebuilt_checksum # Makes sure it is empty before assigning.
+
+# IF there are runfiles, the directory structure will be different.
+# The binaries folder contains different directories for the its artifacts and
+# the artifacts runfiles. Obtain the rebuilt binaries and copy them to the
+# path at root before cleaning up and deleting the repo.
+if [[ "$artifact_name" == *"_deploy.jar"* ]]
+then
+      # Uses _deploy.jar as a field seperator and grabs the field before it.
+      # Directory of Java artifacts is same as run script name.
+      run_script_name=$(echo "$artifact_name" | awk -F'_deploy.jar' '{print $1}')
+      cd $binaries_dir/
+      rebuilt_checksum=$(sha256sum ./"$run_script_name"/"$artifact_name" | awk '{ print $1 }')
+
+      # Copy the entire directory, including the run script.
+      cp -R ./"$run_script_name" ./../../"$rebuilt_artifacts_dir"/
+else
+    if [[ "${NEEDS_RUNFILES}" == "true" ]]
+    then
+        # For non-java targets with runfiles.
+      cd $binaries_dir/
+      rebuilt_checksum=$(sha256sum ./"$artifact_name"/"$artifact_name" | awk '{ print $1 }')
+
+      # Copy entire directory, including the runfiles.
+      cp -R ./"$artifact_name" ./../../"$rebuilt_artifacts_dir"/
```

Comment:
nit: indent consistency
```suggestion
        # For non-java targets with runfiles.
        cd $binaries_dir/
        rebuilt_checksum=$(sha256sum ./"$artifact_name"/"$artifact_name" | awk '{ print $1 }')

        # Copy entire directory, including the runfiles.
        cp -R ./"$artifact_name" ./../../"$rebuilt_artifacts_dir"/
```

Reply:
indented!